app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt

```diff
@@ -86,18 +86,34 @@ private fun MessagePreview.uiLastMessageContent(): UILastMessageContent {
                     userUIText,
                     UIText.PluralResource(
                         R.plurals.last_message_people_added,
-                        (content as WithUser.MembersAdded).count,
-                        (content as WithUser.MembersAdded).count,
-                    )
-                )
-                is WithUser.MembersRemoved -> UILastMessageContent.SenderWithMessage(
-                    userUIText,
-                    UIText.PluralResource(
-                        R.plurals.last_message_people_removed,
-                        (content as WithUser.MembersRemoved).count,
-                        (content as WithUser.MembersRemoved).count
+                        (content as WithUser.MembersAdded).userIdList.size,
+                        (content as WithUser.MembersAdded).userIdList.size,
                     )
                 )
+
+                is WithUser.MembersRemoved -> {
+                    val membersRemovedContent = (content as WithUser.MembersRemoved)
+                    val isSelfRemoved = membersRemovedContent.userIdList.contains(selfUserId)
+                    val isSelfLeft = isSelfMessage && isSelfRemoved
+
+                    when {
+                        isSelfLeft -> UILastMessageContent.TextMessage(
+                            MessageBody(UIText.StringResource(R.string.left_conversation_group_success))
+                        )
+                        isSelfRemoved -> UILastMessageContent.SenderWithMessage(
+                            userUIText,
+                            UIText.StringResource(R.string.last_message_self_removed)
```

Comment:
Bigger problem is here I think, because in this case, someone could remove you and some more people at the same time, so now we would only show that someone removed you but it ignores all others that also were removed 🤔

Reply:
Fixed